dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushPublishActionlet.java

```diff
@@ -48,59 +56,159 @@ public String getHowTo() {
+					 return data;
+				 }
+		 );
+	}
+
+	@Override
+	public void executeBatchAction(final User user,
+			final ConcurrentMap<String, Object> context,
+			final WorkflowActionClass actionClass,
+			final Map<String, WorkflowActionClassParameter> params) throws WorkflowActionFailureException {
+
+		 final Object object = context.get(actionClass.getId());
+		 final PushPublishBatchData pushPublishBatchData = PushPublishBatchData.class.cast(object);
+		 try {
+			 doPushPublish(pushPublishBatchData.getData(), pushPublishBatchData.getIdentifiers(), user);
+		 }catch (Exception e){
+			 Logger.debug(PushPublishActionlet.class, e.getMessage());
+		 	 throw new WorkflowActionFailureException(e.getMessage(), e);
+		 }
+	}
+
+	@Override
+	public List <String> getObjectsForBatch(final ConcurrentMap<String, Object> context,
+			final WorkflowActionClass actionClass){
+		final String actionletInstanceId = actionClass.getId();
+		final Object object = context.get(actionletInstanceId);
+		if(null == object) {
+		   return Collections.emptyList();
+		}
+		final PushPublishBatchData pushPublishBatchData = PushPublishBatchData.class.cast(object);
+		return pushPublishBatchData.getIdentifiers();
+	}
 
-			List<String> identifiers = new ArrayList<String>();
-			identifiers.add(ref.getIdentifier());
 
-			Bundle bundle = new Bundle(null, publishDate, null, processor.getUser().getUserId(), forcePush);
-        	APILocator.getBundleAPI().saveBundle(bundle, envsToSendTo);
 
-			publisherAPI.addContentsToPublish(identifiers, bundle.getId(), publishDate, processor.getUser());
-			if(!_contentPushNeverExpire && (!"".equals(_contentPushExpireDate.trim()) && !"".equals(_contentPushExpireTime.trim()))){
-				Date expireDate = dateFormat.parse(_contentPushExpireDate+"-"+_contentPushExpireTime);
-				bundle = new Bundle(null, publishDate, expireDate, processor.getUser().getUserId(), forcePush);
-            	APILocator.getBundleAPI().saveBundle(bundle, envsToSendTo);
-				publisherAPI.addContentsToUnpublish(identifiers, bundle.getId(), expireDate, processor.getUser());
+	private void doPushPublish(final Map<String,String> pushPublishData, final List<String> identifiers, final User user)
+			throws DotDataException, ParseException, DotPublisherException {
+
+			final String _contentPushPublishDate = pushPublishData.get("wfPublishDate");
+			final String _contentPushPublishTime = pushPublishData.get("wfPublishTime");
+			final String _contentPushExpireDate = pushPublishData.get("wfExpireDate");
+			final String _contentPushExpireTime = pushPublishData.get("wfExpireTime");
+			final boolean _contentPushNeverExpire =
+					"on".equals(pushPublishData.get("wfNeverExpire")) || "true"
+							.equals(pushPublishData.get("wfNeverExpire"));
+			final String whoToSendTmp = pushPublishData.get("whereToSend");
+			final String forcePushStr = pushPublishData.get("forcePush");
+			final boolean forcePush = (forcePushStr != null && forcePushStr.equals("true"));
```

Comment:
you can replace it by "true".equals (forcePushStr)

Reply:
done